flink-runtime/src/main/java/org/apache/flink/runtime/net/NetUtils.java

```diff
@@ -189,9 +191,17 @@ public static InetAddress findConnectingAddress(InetSocketAddress targetAddress,
 		long currentSleepTime = MIN_SLEEP_TIME;
 		long elapsedTime = 0;
 
+		// before trying with different strategies: test with getLocalHost():
+		InetAddress localhostName = InetAddress.getLocalHost();
+
+		if(tryToConnect(localhostName, targetAddress, AddressDetectionState.ADDRESS.getTimeout(), false)) {
+			LOG.debug("Using immediately InetAddress.getLocalHost() for the connecting address");
```

Comment:
These are the produced log statements in `DEBUG` level:
```
16:12:19,822 INFO  org.apache.flink.runtime.util.LeaderRetrievalUtils            - Trying to select the network interface and address to use by connecting to the leading JobManager.
16:12:19,822 INFO  org.apache.flink.runtime.util.LeaderRetrievalUtils            - TaskManager will try to connect for 10000 milliseconds before falling back to heuristics
16:12:19,833 INFO  org.apache.flink.runtime.net.NetUtils                         - Retrieved new target address /10.240.221.7:33378.
16:12:19,835 DEBUG org.apache.flink.runtime.net.NetUtils                         - Trying to connect to (/10.240.221.7:33378) from local address cdh544-master.c.astral-sorter-757.internal/10.240.242.143 with timeout 50
16:12:19,838 DEBUG org.apache.flink.runtime.net.NetUtils                         - Using immediately InetAddress.getLocalHost() for the connecting address
16:12:19,839 INFO  org.apache.flink.runtime.taskmanager.TaskManager              - TaskManager will use hostname/address 'cdh544-master.c.astral-sorter-757.internal' (10.240.242.143) for communication.
16:12:19,839 INFO  org.apache.flink.runtime.taskmanager.TaskManager              - Starting TaskManager in streaming mode BATCH_ONLY
16:12:19,839 INFO  org.apache.flink.runtime.taskmanager.TaskManager              - Starting TaskManager actor system at cdh544-master.c.astral-sorter-757.internal:0
```
I think the messages in `INFO` level contain enough information for users to understand whats going on.